lib/rex/post/meterpreter/packet.rb

```diff
@@ -665,6 +665,44 @@ def initialize(type = nil, method = nil)
     end
   end
 
+  #
+  # Override the function that creates the raw byte stream for
+  # sending so that it generates an XOR key, uses it to scramble
+  # the serialized TLV content, and then returns the key plus the
+  # scrambled data as the payload.
+  #
+  def to_r
```

Comment:
That still doesn't stop one of the middle bytes from being null. If lsb is
0xff then adding 1 results in null. My goal was to force every byte to
change.